src/scope/core/pipelines/defaults.py

```diff
@@ -162,19 +165,15 @@ def prepare_for_mode(
     """
     from .interface import Requirements
 
-    # Calculate video input size if not provided
-    if video_input_size is None:
-        video_input_size = calculate_video_input_size(components_config)
-
-    # If video is explicitly provided, use video mode
+    # Video mode is indicated by presence of 'video' in kwargs
+    # (FrameProcessor sets this based on frontend's input_mode)
     if kwargs.get("video") is not None:
+        # Calculate video input size if not provided
+        if video_input_size is None:
+            video_input_size = calculate_video_input_size(components_config)
```

Comment:
It's a bit confusing to me to see that we're defining a default input size elsewhere but we're basically always calculating the required input size here. If we always calculate the required input size can we just have that be the source of the truth as opposed to also defining a default value elsewhere that is never going to be used? In this case the default value is the calculated value.